mantidproject/mantid
# Remove PyChop2 interface and deprecate PyChop2 import

**Description of work.**
This PR removes the PyChop2 interface from the code as it is no longer used.

It also adds a deprecation warning when someone attempts the following:
```
from mantidqtinterfaces.PyChop import PyChop2
inst = PyChop2("MARI")
```
Instead, users should now use:
```
from pychop.Instruments import Instrument
inst = Instrument("MARI")
```

**To test:**
1. Open Mantid
2. Attempt some of the following commands and observe a deprecation warning that makes sense:
![image](https://user-images.githubusercontent.com/40830825/175330157-cdfd44b7-9709-41d1-9beb-326214ae97e9.png)

3. grep for `PyChop2`. There should only be two __init__ files with a reference to this

Fixes #34102

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
